File: UI/src/app/models/category.ts
import { GameQuestion, QuestionData, QuestionPointer } from "./question";

export interface Category {
    id: string;
    name: string;
    note?: string;
}

export interface CategoryData extends Category {
    questions: Array<QuestionData>;
}

export interface CategoryInstance extends Category {
    questions: Array<QuestionPointer>;
}

export interface GameCategory extends Category {
    questions: Array<GameQuestion>;
}

File: Api/TriviaGame.Api/TriviaGame.Api/Models/Category.cs
using System.ComponentModel.DataAnnotations;

namespace TriviaGame.Api.Models;

public class Category
{
    [Key]
    public string Id { get; set; }
    public string RoundId { get; set; }
    public string Name { get; set; }
    public string? Note { get; set; }
    public IEnumerable<Question> Questions { get; set; }
}
